### DISCUSSION TITLE
Potoprism container stoped after launch

### DISCUSSION THREAD

**Discussion by SergeiAlekseenko** [photoprism/photoprism#3875] [Q&A]
Hello,

After a long photos indexing process Photoprism container stopped and don't want to start again.
I switched on debug mode and trace logging but I don't see any interesting logs to fix the trouble.

How to debug what's going on inside the photoprism container after launch?

> $ docker-compose up
> [+] Running 2/0
>  ✔ Container photoprism-mariadb-1     Running                                                                                                                                                                 0.0s
>  ✔ Container photoprism-photoprism-1  Created                                                                                                                                                                 0.0s
> Attaching to photoprism-mariadb-1, photoprism-photoprism-1
> photoprism-photoprism-1  | started 231011 as root (amd64-prod)
> photoprism-photoprism-1  | Problems? Our Troubleshooting Checklists help you quickly diagnose and solve them:
> photoprism-photoprism-1  | https://docs.photoprism.app/getting-started/troubleshooting/
> photoprism-photoprism-1  | file umask....: "0002" (u=rwx,g=rwx,o=rx)
> photoprism-photoprism-1  | home directory: /photoprism
> photoprism-photoprism-1  | assets path...: /opt/photoprism/assets
> photoprism-photoprism-1  | storage path..: /photoprism/storage
> photoprism-photoprism-1  | config path...: default
> photoprism-photoprism-1  | cache path....: default
> photoprism-photoprism-1  | backup path...: /photoprism/storage/backups
> photoprism-photoprism-1  | import path...: /photoprism/import
> photoprism-photoprism-1  | originals path: /photoprism/originals
> photoprism-photoprism-1  | running as uid 0
> photoprism-photoprism-1  | /opt/photoprism/bin/photoprism start
> photoprism-photoprism-1  | time="2023-11-10T12:43:14Z" level=debug msg="settings: loaded from /photoprism/storage/config/settings.yml"
> photoprism-photoprism-1  | time="2023-11-10T12:43:14Z" level=debug msg="config: running on 'Intel(R) Xeon(R) CPU           E5606  @ 2.13GHz', 17 GB memory detected"
> photoprism-photoprism-1  | time="2023-11-10T12:43:14Z" level=info msg="Become a member today, support our mission and enjoy our member benefits! 💎"
> photoprism-photoprism-1  | time="2023-11-10T12:43:14Z" level=info msg="Visit https://www.photoprism.app/membership to learn more."
> photoprism-photoprism-1  | time="2023-11-10T12:43:14Z" level=debug msg="config: successfully initialized [3.569822ms]"
> photoprism-mariadb-1     | 2023-11-10 12:43:14 11 [Warning] Aborted connection 11 to db: 'photoprism' user: 'photoprism' host: '172.18.0.3' (Got an error reading communication packets)
> photoprism-photoprism-1 exited with code 0

**Comment by lastzero** [comment]
I can't tell from the logs what is causing this. It would be helpful to know more about your configuration and the host operating system environment, for example it could be a Linux kernel security or permission issue:

- https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
- https://docs.photoprism.app/getting-started/troubleshooting/docker/#kernel-security
- https://docs.photoprism.app/getting-started/troubleshooting/docker/#file-permissions

**Reply by lastzero** [reply]
Also verify if there is enough free space on all disks... this is a common problem after indexing as the thumbnails take up space.

**Reply by SergeiAlekseenko** [reply]
Hi, @lastzero 

Thank you for your answers.

**System info**

> $ uname -a
> Linux harmonica.ru 5.4.0-166-generic #183-Ubuntu SMP Mon Oct 2 11:28:33 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

> $ cat /etc/*-release
> DISTRIB_ID=Ubuntu
> DISTRIB_RELEASE=20.04
> DISTRIB_CODENAME=focal
> DISTRIB_DESCRIPTION="Ubuntu 20.04.6 LTS"
> NAME="Ubuntu"
> VERSION="20.04.6 LTS (Focal Fossa)"
> ID=ubuntu
> ID_LIKE=debian
> PRETTY_NAME="Ubuntu 20.04.6 LTS"
> VERSION_ID="20.04"

**AppArmor** is disabled

> $ systemctl status apparmor
> ● apparmor.service - Load AppArmor profiles
>      Loaded: loaded (/lib/systemd/system/apparmor.service; disabled; vendor preset: enabled)
>      Active: inactive (dead)
>        Docs: man:apparmor(7)
>              https://gitlab.com/apparmor/apparmor/wikis/home/

It's enough **space on my disk**

> $ df -h .
> Filesystem      Size  Used Avail Use% Mounted on
> /dev/sdd1       3.6T  1.2T  2.3T  34% /STORAGE2

**Memory**

> $ free -h
>               total        used        free      shared  buff/cache   available
> Mem:           15Gi       6.4Gi       6.7Gi       256Mi       2.6Gi       8.6Gi
> Swap:         4.0Gi          0B       4.0Gi
> 

Photoprism successfully scanned 28k photos and doesn't want to start again after stopping containers.
I can fix it only removing its database folder. It starts well with a fresh database.

**Reply by lastzero** [reply]
In that case, it could be a database configuration or upgrade problem, as the logs look similar to this:

- https://github.com/photoprism/photoprism/discussions/3876

**Reply by lastzero** [reply]
For MariaDB related issues, see:
- https://docs.photoprism.app/getting-started/troubleshooting/mariadb/#version-upgrade

**Reply by SergeiAlekseenko** [reply]
I use the latest photoprism image and mariadb:10.11 without any parameters changed except of db params (https://dl.photoprism.app/docker/docker-compose.yml).

I restarted rescan again with a clean database, but I believe I'll reproduce the issue after it has scanned more than 20k images.

It would be great if you could add any additional diagnostic data to the log to understand the issue.

**Reply by lastzero** [reply]
It seems that PhotoPrism can't properly access MariaDB. If this were a general problem, no one would be able to use PhotoPrism, so I assume it must be something specific to your setup. For example, what kind of storage do you use for MariaDB?

**Reply by SergeiAlekseenko** [reply]
I agee, PhotoPrism starts and works fine the first time. The issue appears when I stop and start docker-compose again after indexing photos.
I use docker-compose with two containers that specified by default: PhotoPrism and MariaDB.  As a storage I use my internal HDD. 

If Photoprism coudn't access MariaDB we didn't see MaridDB logs, but they are:
```
photoprism-photoprism-1 | time="2023-11-10T12:43:14Z" level=debug msg="config: successfully initialized [3.569822ms]"
photoprism-mariadb-1 | 2023-11-10 12:43:14 11 [Warning] Aborted connection 11 to db: 'photoprism' user: 'photoprism' host: '172.18.0.3' (Got an error reading communication packets)
photoprism-photoprism-1 exited with code 0
```
Something happened in PhotoPrism after the connection I think.

At the moment it is working and indexing photos. I will try to restart the containers when the scanning process finishes.
